src/Core/LongRunning/LongRunningOperation.php

```diff
@@ -0,0 +1,377 @@
+     *        operation.metadata.typeUrl.
+     */
+    public function __construct(
+        LongRunningConnectionInterface $connection,
+        $name,
+        array $callablesMap
+    ) {
+        $this->connection = $connection;
+        $this->name = $name;
+        $this->callablesMap = $callablesMap;
+    }
+
+    /**
+     * Return the Operation name.
+     *
+     * Example:
+     * ```
+     * $name = $operation->name();
+     * ```
+     *
+     * @return string
+     */
+    public function name()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Check if the Operation is done.
+     *
+     * If the Operation state is not available, a service request may be executed
+     * by this method.
+     *
+     * Example:
+     * ```
+     * if ($operation->done()) {
+     *     echo "The operation is done!";
+     * }
+     * ```
+     *
+     * @param array $options [optional] Configuration options.
+     * @return bool
+     */
+    public function done(array $options = [])
+    {
+        return (isset($this->info($options)['done']))
+            ? $this->info['done']
+            : false;
+    }
+
+    /**
+     * Get the state of the Operation.
+     *
+     * Return value will be one of `LongRunningOperation::STATE_IN_PROGRESS`,
+     * `LongRunningOperation::STATE_SUCCESS` or
+     * `LongRunningOperation::STATE_ERROR`.
+     *
+     * If the Operation state is not available, a service request may be executed
+     * by this method.
+     *
+     * Example:
+     * ```
+     * switch ($operation->state()) {
+     *     case LongRunningOperation::STATE_IN_PROGRESS:
+     *         echo "Operation is in progress";
+     *         break;
+     *
+     *     case LongRunningOperation::STATE_SUCCESS:
+     *         echo "Operation succeeded";
+     *         break;
+     *
+     *     case LongRunningOperation::STATE_ERROR:
+     *         echo "Operation failed";
+     *         break;
+     * }
+     * ```
+     *
+     * @param array $options [optional] Configuration options.
+     * @return string
+     */
+    public function state(array $options = [])
+    {
+        if (!$this->done($options)) {
+            return self::STATE_IN_PROGRESS;
+        }
+
+        if ($this->done() && $this->result()) {
+            return self::STATE_SUCCESS;
+        }
+
+        return self::STATE_ERROR;
+    }
+
+    /**
+     * Get the Operation result.
+     *
+     * The return type of this method is dictated by the type of Operation.
+     *
+     * Returns null if the Operation is not yet complete, or if an error occurred.
+     *
+     * If the Operation state is not available, a service request may be executed
+     * by this method.
+     *
+     * Example:
+     * ```
+     * $result = $operation->result();
+     * ```
+     *
+     * @param array $options [optional] Configuration options.
+     * @return mixed|null
+     */
+    public function result(array $options = [])
```

Comment:
The convention in our library is to expose the entity response data via a method called `info()`. In this case, calling `$result->info()` would return the [long running operation data](https://cloud.google.com/spanner/docs/reference/rpc/google.longrunning#google.longrunning.Operation), which includes the metadata.